docs/proposals/PROPOSAL_F.md

```diff
@@ -0,0 +1,374 @@
+Those annotations **SHOULD** be added to the referenced image descriptor.
+The annotations **MAY** store small data.
+The list will evolve depending adoption.
+
+Example of well-known annotations, adding cosign signatures to an image:
+```json
+{
+   "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
+   "digest": "sha256:c3c58223e2af75154c4a7852d6924b4cc51a00c821553bbd9b3319481131b2e0",
+   "size": 528,
+   "platform": {
+      "architecture": "arm64",
+      "os": "linux"
+   },
+   "annotations": {
+      "dev.cosignproject.cosign/signature.e59879c": "...",
+      "dev.sigstore.cosign/bundle.6436749": "...",
+      "dev.sigstore.cosign/certificate.776a3f1": "...",
+      "dev.sigstore.cosign/chain.8adf60e": "...",
+      "dev.cosignproject.cosign/signature.de25bf6": "...",
+      "dev.sigstore.cosign/bundle.37d34ff": "...",
+      "dev.sigstore.cosign/certificate.e8adf60": "...",
+      "dev.sigstore.cosign/chain.b1c64a3": "...",
+      "org.opencontainers.signatures.index":"dev.cosignproject.cosign/signature.e59879c,dev.cosignproject.cosign/signature.de25bf6,dev.sigstore.cosign/bundle.37d34ff,dev.sigstore.cosign/certificate.e8adf60,dev.sigstore.cosign/chain.b1c64a3,dev.sigstore.cosign/bundle.6436749,dev.sigstore.cosign/certificate.776a3f1,dev.sigstore.cosign/chain.8adf60e",
+   }
+}
+```
+
+### Scenarii
+
+The current proposal doesn't change the workflow for [the Refrigerated Truck Driver 🚚](https://github.com/opencontainers/wg-reference-types/blob/main/docs/PERSONAS.md#3-the-refrigerated-truck-driver-), [the Bodega Owner 🏪](https://github.com/opencontainers/wg-reference-types/blob/main/docs/PERSONAS.md#4-the-bodega-owner-) and the [The Ice Cream Lover 😍](https://github.com/opencontainers/wg-reference-types/blob/main/docs/PERSONAS.md#5-the-ice-cream-lover-) personas.
+Two scenarii are identified for the [Ice Cream Factory Worker 🏭](https://github.com/opencontainers/wg-reference-types/blob/main/docs/PERSONAS.md#2-the-ice-cream-factory-worker-) and [Health Inspector 🕵️‍♀️](https://github.com/opencontainers/wg-reference-types/blob/main/docs/PERSONAS.md#2-the-ice-cream-factory-worker-) personas.
+
+1/ The 🏭 wants to enhance an existing image without changing its digest, already consumed by 😍, so 🕵️ can validate it.
+2/ The 🏭 pushes an image with all the pieces needed by 🕵 without changing 😍 workflow.
+
+#### Enhance existing image
+
+An image is already pushed and used. To add artifacts and annotations to it without re-pushin and changing the known digest, a new OCI index is pushed on a new tag, referencing the original image.
+The tag **SHOULD** have the following format:
+`<repo>:<alg>-<ref>.<hash>`:
+- E.g. `registry.example.org/project-d:sha256-0000000000000000000000000000000000000000000000000000000000000000.0404040404040404`
+- `<alg>`: the digest algorithm
+- `<ref>`: the referenced digest (limit of 64 characters)
+- `<hash>`: hash of this artifact (limit of 16 characters)
```

Comment:
Is the artifact hash still needed with this proposal? I thought that was going away to avoid the tag listing.

Reply:
It's needed when you would like to add metadata to an existing object.

As the proposal is that you append metadata to the OCI index, you do not need to list tags. You only need to have your unsigned existing object's digest and then you can compute the tag for the index with metadata.

Reply:
Is that mixing up the reference digest with the artifact hash? The original index you want to extend is the first digest. The digest of the new index with the modified fields is the second hash, and that isn't known by consumers in advance so they need to use a tag list to find them.

Reply:
Distribution spec doesn't seem to have a way to pull/push index manifests. Would the new OCI index be a nested index?

Reply:
@nishakm 
> Distribution spec doesn't seem to have a way to pull/push index manifests
I'm not a distribution spec expert but I believe they're the same as other [manifests](https://github.com/opencontainers/distribution-spec/blob/dd38b7ed8a995fc2f6e730a4deae60e2c0ee92fe/spec.md#pulling-manifests)

There's a useful diagram [here](https://github.com/google/go-containerregistry/tree/d7f8d06c87ed209507dd5f2d723267fe35b38a9f/pkg/v1/remote#anatomy-of-an-image-upload) to see what's happening.

@sudo-bmitch 
> That would mean there's only one digest tag for each referenced digest, pointing to the extended index, negating the need for the additional <hash> entry.

Yes, I believe that you're right. I didn't read it closely enough the first time.

Reply:
@sudo-bmitch yes totally, removing it 👍

Reply:
Fixed https://github.com/opencontainers/wg-reference-types/pull/52/commits/47a0daa6a00ac31c25ba0d180c158ff9442e6634